controllers/complianceeventsapi/server.go

```diff
@@ -471,6 +505,87 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
 		}
 	}
 
+	parsed, err := setAuthorizedClusters(ctx, db, parsed, userConfig)
+	if err != nil {
+		// ErrNoAccess needs queryOptions
+		return parsed, err
+	}
+
+	return parsed, nil
+}
+
+// getAuthorizedClusterNames verifies that if a cluster cfilter is provided,
+// the user has access to this filter. If no cluster filter is provided,
+// it sets the cluster filter to all managed clusters the user has access to.
+func setAuthorizedClusters(ctx context.Context, db *sql.DB, parsed *queryOptions,
+	userConfig *rest.Config,
+) (*queryOptions, error) {
+	unAuthorizedClusters := []string{}
+
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return parsed, err
+	}
+
+	clusterIDs, okClusterIds := parsed.Filters["clusters.cluster_id"]
+	if okClusterIds {
+		// Convert id to name
+		for _, id := range clusterIDs {
+			clusterName, err := getClusterNameFromID(ctx, db, id)
+			if err != nil {
```

Comment:
This is too generic of a check. I think `getClusterNameFromID` should return an empty string when it can't find the cluster name from the cluster ID. I think the scan will return something like row not found. Then if clusterName is an empty string, it'll come back as unauthorized. If there is an error returned, then it's an internal error (e.g. Database query failed).

Reply:
How about this 
```
if errors.Is(err, sql.ErrNoRows) {
					// This leads to send forbidden error to the user
					unAuthorizedClusters = append(unAuthorizedClusters, id)
				}
```